admin/updates/update-4.sql

```diff
@@ -0,0 +1,28 @@
+BEGIN;
+
+ALTER TABLE dataset_snapshot RENAME TO dataset_eval_sets;
+ALTER TABLE dataset_eval_sets RENAME CONSTRAINT dataset_snapshot_pkey TO dataset_eval_sets_pkey;
+
+CREATE TABLE dataset_snapshot (
+  id         UUID, -- PK
+  dataset_id UUID, -- FK to dataset
+  data       JSONB                    NOT NULL,
+  created    TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW()
+);
+
+ALTER TABLE dataset_snapshot ADD CONSTRAINT dataset_snapshot_pkey PRIMARY KEY (id);
+
+ALTER TABLE dataset_snapshot
+  ADD CONSTRAINT dataset_id_fk_dataset
+  FOREIGN KEY (dataset_id)
+  REFERENCES dataset (id);
+
+DELETE FROM dataset_eval_jobs;
```

Comment:
we discussed this in person. This update script should be modified to create a snapshot for all existing datasets and update the columns in `dataset_eval_jobs` to point to the right things